asammdf/mdf_v4.py

```diff
@@ -433,24 +433,28 @@ def _read(self):
                         channel_group['flags'] &= ~v4c.FLAG_CG_BUS_EVENT
                     else:
                         comment = channel_group.comment.replace(' xmlns="http://www.asam.net/mdf/v4"', '')
-                        comment_xml = ET.fromstring(comment)
-                        can_msg_type = comment_xml.find('.//TX').text.strip(' \t\r\n')
-                        if can_msg_type == 'CAN_DataFrame':
-                            common_properties = comment_xml.find(".//common_properties")
-                            can_id = 1
-                            message_id = -1
-                            for e in common_properties:
-                                name = e.get('name')
-                                if name == 'MessageID':
-                                    if e.get('ci') is not None:
-                                        can_id = int(e.get('ci'))
-                                    message_id = int(e.text)
-                            grp['can_id'] = can_id
-                            grp['message_name'] = message_name
-                            grp['message_id'] = message_id
+                        try:
+                            comment_xml = ET.fromstring(comment)
+                            text = comment_xml.find('.//TX').text
+                            can_msg_type = text.strip(' \t\r\n') if text else text
+                            if can_msg_type == 'CAN_DataFrame':
+                                common_properties = comment_xml.find(".//common_properties")
+                                can_id = 1
+                                message_id = -1
+                                for e in common_properties:
+                                    name = e.get('name')
+                                    if name == 'MessageID':
+                                        if e.get('ci') is not None:
+                                            can_id = int(e.get('ci'))
+                                        message_id = int(e.text)
+                                grp['can_id'] = can_id
+                                grp['message_name'] = message_name
+                                grp['message_id'] = message_id
 
-                        else:
-                            warnings.warn('Invalid bus logging channel group metadata: {}'.format(comment))
+                            else:
+                                warnings.warn('Invalid bus logging channel group metadata: {}'.format(comment))
+                                channel_group['flags'] &= ~v4c.FLAG_CG_BUS_EVENT
+                        except ET.ParseError as e:
```

Comment:
I would leave out the "as e" assignment if it's not used in the exception block.